library/ssl_cli.c

```diff
@@ -4236,13 +4236,23 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
     switch( ssl->state )
     {
         case MBEDTLS_SSL_HELLO_REQUEST:
+            mbedtls_platform_random_delay();
```

Comment:
I'm afraid that if you want this double check to be effective, the ssl state has to be declared as volatile.